relay/http.go

```diff
@@ -209,7 +229,11 @@ func (h *HTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		b := b
 		go func() {
 			defer wg.Done()
-			resp, err := b.post(outBytes, query, authHeader)
+			q := false
+			if reqPath == "/query" {
```

Comment:
checking this kind of things multiple times throughout the code further reinforces that a re-structure is needed, and this PR is not a suitable implementation